Delete Workbench workspace QMenu after use

martyngigg opened the pull request:
**Description of work.**

This is not a large leak but it can cause
confusion when trying to decide with methods
like `QApplication.topLevelWindows()` whether
everything that was to be deleted has been.

**To test:**

To check you understand the problem run the instructions before merging and then after:

* Start workbench
* Run `CreateSampleWorkspace`
* Right-click on the workspace
* In the script editor run:
```python
from qtpy.QtWidgets import QApplication
import pprint

pprint.pprint([w.objectName() for w in QApplication.topLevelWindows()])
```

The list should not contain any references to `WorkspaceContextMenu` after the fix.

*There is no associated issue. It was disocvered while debugging #33662*

*This does not require release notes* because **a complete summary once #33662 and #33661 is fixed will be included.**

<!-- Ensure the base of this PR is correct (e.g. release-next or master)
Finally, don't forget to add the appropriate labels, milestones, etc.!  -->

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

martyngigg committed:
Delete workspace QMenu after use

This is not a large leak but it can cause
confusion when trying to decide with methods
like QApplication.topLevelWindows() whether
everything that was to be deleted has been

martyngigg committed:
Fix CppCheck warnings in tree widget

thomashampson reviewed (commented):
The `plotSubMenu` (created on line 121) is only added to the `menu` is the workspace is not single valued, otherwise it is just left to dangle (see line 159). This isn't a massive issue since it's given `menu` as a parent when it's constructed, but I wonder if the logic can be tidied up anyway to avoid confusion.

martyngigg committed:
Split large method into methods per workspace type

Each method is self-contained on the workspace type
so the methods that will appear can be understood
much more easily.

thomashampson reviewed (approved):
Fixes the leak, refactors the code to make it more readable and resolves a few cppcheck defects. 👍

martyngigg commented:
@mantidproject/gatekeepers Could we get this merged?
